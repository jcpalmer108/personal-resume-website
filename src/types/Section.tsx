import { ReactNode } from 'react'

export type SectionProps = {
  children: ReactNode,
  label?: string,
  light?: boolean,
  thin?: boolean,
  wide?: boolean,
  center?: boolean,
  bottom?: boolean,
  noTop?: boolean,
  noBorder?: boolean
}

export type WrapperProps = {
  light?: boolean,
  thin? : boolean,
  bottom?: boolean,
  noBorder?: boolean
}

export type HeaderProps = {
  center?: boolean,
  noTop?: boolean
}

export type ChildrenProps = {
  padded?: boolean
}
