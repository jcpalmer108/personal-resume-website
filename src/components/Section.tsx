import { 
  Wrapper, 
  Header, 
  Line, 
  Label, 
  Title, 
  Children
} from "../styles/components/Section";
import line from "../assets/images/label-line.svg";
import { SectionProps } from "../types/Section"

export default function Section({ 
  label,
  children,
  light,
  thin,
  wide,
  center,
  bottom,
  noTop,
  noBorder
}: SectionProps) {
  return (
    <Wrapper noBorder={noBorder} light={light} thin={thin} bottom={bottom} data-testid="Wrapper">
      {label && (
        <Header noTop={noTop} data-testid="Header" center={center} >
          <Label data-testid="Label">
            <Line src={line} alt="test"/>
            <Title>{label.toUpperCase()}</Title>
          </Label>
        </Header>
      )}
      <Children padded={!wide} data-testid="Children">{children}</Children>
    </Wrapper>
  )
}