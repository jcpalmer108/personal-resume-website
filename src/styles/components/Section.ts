import styled from "styled-components";
import { desktopBreakpoint, contentWidth, silver_polish, tabletBreakpoint } from "../variables";
import { WrapperProps, HeaderProps, ChildrenProps } from "../../types/Section"

export const Wrapper = styled.div<WrapperProps>`
  max-width: ${contentWidth}px;
  margin: auto;
  border-left: solid;
  border-right: solid;
  border-bottom: solid;
  border-color: transparent;
  border-width: 0px;

  @media (min-width: ${contentWidth}px) {
    ${props => props.noBorder ? `
      border: none;
    ` : `
      border-bottom: ${props.bottom ? "soild" : "none" };
      border-color: ${props.light ? "white" : silver_polish};
      border-width: ${props.thin ? 0.2 : 1}px;    
    `}
  }
`

export const Header = styled.div<HeaderProps>`
  width: fit-content;
  margin: ${props => props.center ? "auto" : "0px"};
  padding-top: 50px; 
  padding-left: 16px;
  
  @media (min-width: ${tabletBreakpoint}px) {
    margin: 0px;
    padding-left: 25px;
    padding-top: ${props => props.noTop ? '25px' : '80px'};
  }

  @media (min-width: ${desktopBreakpoint}px) {
    padding-left: 60px;
    padding-top: 85px;
  }
`

export const Label = styled.div`
  display: flex;
  column-gap: 10px;
  font-size: 11px;
  font-family: Gilroy-Regular;

  @media (min-width: ${tabletBreakpoint}px) {
    transform: rotate(-90deg);
    padding-bottom: calc(100% - 15px);
    font-size: 14px;
  }

  @media (min-width: ${desktopBreakpoint}px) {
    padding-bottom: calc(100% - 17px);
    font-size: 16px;
  }

`

export const Children = styled.div<ChildrenProps>`
  padding: ${props => props.padded ? "20px 15px" : "0px"};

  @media (min-width: ${tabletBreakpoint}px) {
    padding: ${props => props.padded ? "40px 25px" : "0px"};
  }

  @media (min-width: ${desktopBreakpoint}px) {
    padding: ${props => props.padded ? "45px 60px 55px 60px" : "0px"};
  }
`


export const Title = styled.span`
  font-size: 12px;

  @media (min-width: ${tabletBreakpoint}px) {
    font-size: 14px;
  }

  @media (min-width: ${desktopBreakpoint}px) {
    font-size: 16px;
  }

`

export const Line = styled.img`
  width: 25px;
`

