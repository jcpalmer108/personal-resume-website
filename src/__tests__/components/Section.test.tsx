/* eslint-disable testing-library/no-node-access */
import { render, screen } from "@testing-library/react";
import "jest-styled-components";
import Section from "../../components/Section";

// params for components
const params = {
  required: {
    children: "hello world",
  },
  optional: {
    label: "test label",
    light: true,
    thin: true,
    wide: true,
    center: true,
    bottom: true,
    noTop: true,
    noBorder: true,
  },
};

describe("Section", () => {
  test("renders if required and optional params are passed in", () => {
    // given
    const { required, optional } = params;
    render(
      <Section
        label={optional.label}
        light={optional.light}
        thin={optional.thin}
        wide={optional.wide}
        center={optional.center}
        bottom={optional.bottom}
        noTop={optional.noTop}
        noBorder={optional.noBorder}
      >
        {required.children}
      </Section>
    );

    // then
    expect(screen.getByTestId("Wrapper")).toMatchSnapshot();
    expect(screen.getByTestId("Wrapper").childElementCount).toEqual(2);
    expect(screen.getByTestId("Label")).toHaveTextContent(
      optional.label.toUpperCase()
    );
    expect(screen.getByTestId("Children")).toHaveTextContent(required.children);
  });

  test("renders if only required params are passed in", () => {
    // given
    const { required } = params;
    render(<Section>{required.children}</Section>);

    // then
    expect(screen.getByTestId("Wrapper")).toMatchSnapshot();
    expect(screen.getByTestId("Wrapper").childElementCount).toEqual(1);
    expect(screen.queryAllByTestId("Label")).toHaveLength(0);
    expect(screen.getByTestId("Children")).toHaveTextContent(required.children);
  });
});
