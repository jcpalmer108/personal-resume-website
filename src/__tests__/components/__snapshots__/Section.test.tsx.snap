// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section renders if only required params are passed in 1`] = `
.c0 {
  max-width: 1155px;
  margin: auto;
  border-left: solid;
  border-right: solid;
  border-color: transparent;
  border-width: 0px;
}

.c1 {
  padding: 20px 15px;
}

@media (min-width:1155px) {
  .c0 {
    border-color: #c6c6c6;
    border-width: 1px;
  }
}

@media (min-width:1155px) {
  .c0 {
    border-bottom: none;
  }
}

@media (min-width:720px) {
  .c1 {
    padding: 40px 25px;
  }
}

@media (min-width:1024px) {
  .c1 {
    padding: 45px 60px 55px 60px;
  }
}

<div
  class="c0"
  data-testid="Wrapper"
>
  <div
    class="c1"
    data-testid="Children"
  >
    <p>
      Child 1
    </p>
  </div>
</div>
`;

exports[`Section renders if required and optional params are passed in 1`] = `
.c0 {
  max-width: 1155px;
  margin: auto;
  border-left: solid;
  border-right: solid;
  border-color: transparent;
  border-width: 0px;
}

.c1 {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  margin: auto;
  padding-top: 50px;
  padding-left: 16px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-column-gap: 10px;
  column-gap: 10px;
  font-size: 11px;
  font-family: Gilroy-Regular;
}

.c5 {
  padding: 0px;
}

.c4 {
  font-size: 12px;
}

.c3 {
  width: 25px;
}

@media (min-width:1155px) {
  .c0 {
    border-color: white;
    border-width: 0.2px;
  }
}

@media (min-width:1155px) {
  .c0 {
    border-bottom: none;
  }
}

@media (min-width:720px) {
  .c1 {
    margin: 0px;
    padding-left: 25px;
    padding-top: 80px;
  }
}

@media (min-width:1024px) {
  .c1 {
    padding-left: 60px;
    padding-top: 85px;
  }
}

@media (min-width:720px) {
  .c2 {
    -webkit-transform: rotate(-90deg);
    -ms-transform: rotate(-90deg);
    transform: rotate(-90deg);
    padding-bottom: calc(100% - 15px);
    font-size: 14px;
  }
}

@media (min-width:1024px) {
  .c2 {
    padding-bottom: calc(100% - 17px);
    font-size: 16px;
  }
}

@media (min-width:720px) {
  .c5 {
    padding: 0px;
  }
}

@media (min-width:1024px) {
  .c5 {
    padding: 0px;
  }
}

@media (min-width:720px) {
  .c4 {
    font-size: 14px;
  }
}

@media (min-width:1024px) {
  .c4 {
    font-size: 16px;
  }
}

<div
  class="c0"
  data-testid="Wrapper"
>
  <div
    class="c1"
    data-testid="Header"
  >
    <div
      class="c2"
      data-testid="Label"
    >
      <img
        alt="test"
        class="c3"
        src="label-line.svg"
      />
      <span
        class="c4"
      >
        TESTING
      </span>
    </div>
  </div>
  <div
    class="c5"
    data-testid="Children"
  >
    <p>
      Child 1
    </p>
  </div>
</div>
`;
